src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Source/SourceUpdaterStep.cs

```diff
@@ -159,6 +161,13 @@ public async Task GetDiagnosticsAsync(CancellationToken token)
             }
         }
 
+        public static Project AddMetadataReferences(Project project)
+        {
+            project = Extensions.Default.CSharp.Analyzers.ApiControllerAnalyzer.AddMetadataReferences(project);
```

Comment:
The source updater step shouldn't depend on extensions (not even the default ones). I'm actually surprised that it has a transitive reference to that package.

Reply:
Maybe I'm too attached to the word "requirement", maybe the SourceUpdater step should check to see if the analyzer needs to be initialized and adding the metadataReference can be part of initialization